### ISSUE TITLE
Disenchanting items when claiming

### ISSUE THREAD

**Issue by donteatlemons** [WasabiThumb/xclaim#4]
On Spigot 1.19, enchanted items in the player's inventory are getting disenchanted when claiming land.

**Comment by WasabiThumb** [comment]
Can you give an example/steps to reproduce? A video would help too. I don't see how this is possible since the entire data of the items are being serialized. This plugin has been in use on a large SMP for a month or so now with no such reports.

**Comment by WasabiThumb** [comment]
Ah, actually, I forgot that on Spigot this is done differently than Paper, which maybe could make it possible for enchantments to be lost. I don't see how though
```java
    @Override
    public byte @NotNull [] itemStackSerializeBytes(@NotNull ItemStack is) {
        Map<String, Object> map = is.serialize();
        ByteArrayOutputStream bos = new ByteArrayOutputStream();
        try (ObjectOutputStream oos = new ObjectOutputStream(bos)) {
            oos.writeInt(map.size());
            for (Map.Entry<String, Object> entry : map.entrySet()) {
                String key = entry.getKey();
                byte[] keyBytes = key.getBytes(StandardCharsets.UTF_8);
                oos.writeInt(keyBytes.length);
                oos.write(keyBytes);
                oos.writeObject(entry.getValue());
            }
        } catch (IOException ignored) { }
        return bos.toByteArray();
    }

    @Override
    public @NotNull ItemStack itemStackDeserializeBytes(byte @NotNull [] bytes) {
        Map<String, Object> map = new HashMap<>();
        ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
        try (ObjectInputStream ois = new ObjectInputStream(bis)) {
            int size = ois.readInt();
            for (int i=0; i < size; i++) {
                int len = ois.readInt();
                String key = new String(ois.readNBytes(len), StandardCharsets.UTF_8);
                Object value = ois.readObject();
                map.put(key, value);
            }
        } catch (IOException | ClassNotFoundException ignored) { }
        return ItemStack.deserialize(map);
    }
    ```
    May I ask why you use Spigot instead of Paper?

**Comment by WasabiThumb** [comment]
I created a [new branch](https://github.com/WasabiThumb/xclaim/tree/4-disenchanting-items-when-claiming) for this issue and made some changes that should help, try downloading [this JAR](https://wasabicodes.xyz/cdn/11e7ca8c8e7dcc2adfea6dfbb942cd0b/xclaim-1.7.0.jar) and see if your issue is fixed.

**Comment by WasabiThumb** [comment]
Appears to be patched in the version provided, I will close this issue and merge in 24 hours unless necessary
[See video here](https://wasabicodes.xyz/cdn/0d7e2a2719d504f9a9ab0f01a907384a/patch.mp4)

**Comment by WasabiThumb** [comment]
Fixed in 1.7.1
